src/base/static/components/organisms/map-measurement-overlay.tsx

```diff
@@ -0,0 +1,369 @@
+/** @jsx jsx */
+import * as React from "react";
+import { css, jsx } from "@emotion/core";
+import styled from "@emotion/styled";
+import PropTypes from "prop-types";
+import { featureCollection, point, lineString, polygon } from "@turf/helpers";
+import area from "@turf/area";
+import distance from "@turf/distance";
+import { geoPath, geoTransform } from "d3-geo";
+import WebMercatorViewport from "viewport-mercator-project";
+import { CanvasOverlay } from "react-map-gl";
+
+import { RegularText } from "../atoms/typography";
+import {
+  CreatePolygonIcon,
+  CreatePolylineIcon,
+  DeleteGeometryIcon,
+  UndoIcon,
+} from "../atoms/icons";
+
+const MeasurementToolIcon = styled("span")(props => ({
+  border: "2px solid transparent",
+  borderRadius: "4px",
+  padding: "4px 4px 0 4px",
+  marginRight: "8px",
+  backgroundColor: props.isSelected ? "#999" : "initial",
+
+  "&:hover": {
+    borderColor: props.isEnabled ? "#999" : "transparent",
+    cursor: props.isEnabled ? "pointer" : "unset",
+  },
+}));
+
+MeasurementToolIcon.defaultProps = {
+  isEnabled: true,
+};
+
+const FEET_PER_MILE = 5280;
+const SQUARE_METERS_PER_ACRE = 0.000247105;
+const measurementFormatter = new Intl.NumberFormat("en-US");
+const MIN_POSITIONS = {
+  "create-polygon": 4,
+  "create-polyline": 2,
+};
+
+export default class MapMeasurementOverlay extends React.Component {
```

Comment:
Is there a reason why we are using class components, and not function components with hooks?

Reply:
It was a little easier for me to think of this as a class component with lifecycle methods. I'll attempt to build it with hooks though...

Reply:
Sounds good, if it's too much of a learning curve all at once.

I just wanted to clarify that all things being equal, we should be writing all new components as functional components.

Reply:
refactored to use hooks